File: 프로그래머스/lv2/구명보트.py
from collections import deque

def solution(people, limit):
    people=deque(sorted(people,reverse=True))
    answer = 0
    while len(people)>1:
        if people[0]+people[-1]<=limit:
            answer+=1
            people.pop()
            people.popleft()
        else:
            answer+=1
            people.popleft()
    if people:
        answer+=1
    return answer
        
        

print(solution([70, 50, 80, 50]	,100))

File: 프로그래머스/lv2/구명보트.java
package lv2;

import java.util.Arrays;
public class 구명보트 {
    public int solution(int[] people, int limit) {
        int answer = 0;
        Arrays.sort(people);
        int index=0;
        for (int i=people.length-1;i>=index;i--){
            if(people[i]+people[index]<=limit){
                index++;
                answer++;
            }
            else answer++;
        }
        return answer;
    }

    public static void main(String[] args) {
        구명보트 boat = new 구명보트();
        System.out.println(boat.solution(new int[]{70, 50, 80, 50},100));
    }
}